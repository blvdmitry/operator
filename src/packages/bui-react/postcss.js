// Re-export of the config provided by bui-postcss package
const config = require("@bookingcom/bui-postcss");

module.exports = config.default;
