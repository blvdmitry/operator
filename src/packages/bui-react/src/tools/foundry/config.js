import React from "react";
import { configure } from "@bookingcom/foundry-react";
import BUIProvider from "../../components/BUIProvider";
import Container from "../../components/Container";
import LegacyTheme from "../../themes/legacy";
import TravellerTheme from "../../themes/traveller";
import TravellerExTheme from "../../themes/traveller_ex";
import RentalcarsTheme from "../../themes/rentalcars";

const Wrapper = ({ children, theme, mode }) => {
  return (
    <BUIProvider
      theme={theme}
      themeMode={mode}
      id="foundry-preview"
      // experiments={{ slider_container_animation_duration: () => true }}
    >
      <Container>{children}</Container>
    </BUIProvider>
  );
};

configure({
  name: "BUI React",
  description: "Component library implementing BUI in React",
  ownership: {
    orgId: [60008228],
    include: ["dbelyaev"],
  },
  url: "https://design-systems.pages.booking.com/bui/bui-web/react/revs/latest/",
  logo: "./public/logo.svg",
  showExamplesInDocsOnly: true,
  themes: [
    {
      theme: TravellerTheme,
      definition: {
        darkDefinition: TravellerTheme.darkDefinition,
        lightDefinition: TravellerTheme.lightDefinition,
      },
    },
    {
      theme: RentalcarsTheme,
      definition: {
        darkDefinition: RentalcarsTheme.darkDefinition,
        lightDefinition: RentalcarsTheme.lightDefinition,
      },
    },
    {
      theme: LegacyTheme,
      definition: {
        darkDefinition: LegacyTheme.darkDefinition,
        lightDefinition: LegacyTheme.lightDefinition,
      },
    },
    {
      theme: TravellerExTheme,
      definition: {
        darkDefinition: TravellerExTheme.darkDefinition,
        lightDefinition: TravellerExTheme.lightDefinition,
      },
    },
  ],
  links: {
    slack: {
      name: "#bui-react",
      url: "https://booking.slack.com/archives/CBBUC8HH7",
    },
    gitlab: {
      name: "BUI React",
      url: "https://gitlab.booking.com/design-systems/bui/bui-web",
    },
  },

  order: [
    {
      name: "Getting started",
      order: 0,
      children: [
        {
          name: "Repository",
          order: 0,
        },
        {
          name: "Iconography",
          order: 1,
        },
        {
          name: "Installation",
          order: 2,
        },
        {
          name: "Major releases",
          order: 3,
        },
        {
          name: "Hooks",
          order: 4,
        },
      ],
    },
    {
      name: "Foundations",
      order: 1,
      children: [
        {
          name: "Color",
          order: 0,
        },
        {
          name: "Spacing",
          order: 1,
        },
        {
          name: "Typography",
          order: 2,
        },
        {
          name: "Other tokens",
          order: 3,
        },
      ],
    },
    {
      name: "Components",
      order: 2,
      children: [
        {
          name: "Utilities",
          order: 0,
        },
        {
          name: "Elements",
          order: 1,
        },
        {
          name: "Containers",
          order: 2,
        },
        {
          name: "Patterns",
          order: 3,
        },
        {
          name: "Deprecated",
          order: 4,
        },
      ],
    },
  ],
  // versioning: true, enable this when we fix versioner issue
  gaId: "UA-92977044-52",
  render: {
    Wrapper,
  },
});
