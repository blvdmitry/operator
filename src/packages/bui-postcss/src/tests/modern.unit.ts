import fs from "fs";
import postcss from "postcss";
import config from "../postcss.config";

test("modern config", async () => {
  const modernConfig = config.modern();
  const input = fs.readFileSync(`${__dirname}/styles.css`);
  const output = await postcss(modernConfig.plugins).process(input, {
    from: "__placeholder-name.css",
  });

  expect(output.css).toMatchSnapshot();
});
