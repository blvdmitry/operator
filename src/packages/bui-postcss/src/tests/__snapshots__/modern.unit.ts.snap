// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modern config 1`] = `
"/* media size definitions */
/* fonts definitions */
:root {
  --color: red;
  --unit: 10px;
  --unit-large: 20px
}
.foo {
  color: var(--color);
  margin-left: var(--unit);
}
.bar {
  color: var(--color);
  margin-left: var(--unit);
}
@media (max-width: 575px) {
  .foo {
    margin-left: var(--unit-large);
  }
}
.bui {
  background: var(--bui_color_brand_primary_background);
  font-size: var(--bui_font_body_1_font-size);
  font-weight: var(--bui_font_body_1_font-weight);
  line-height: var(--bui_font_body_1_line-height);
  font-family: var(--bui_font_body_1_font-family);
}
"
`;
